dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/NetworkApiLiveTest.java

```diff
@@ -65,36 +77,114 @@ public void testCreatePortList() {
+   public void testGetPortList() {
+      FirewallRuleTarget.PortList portList = api().getPortList(portListId);
+      assertNotNull(portList);
+      assertTrue(portList.port().get(0).begin() == 22);
+   }
+
+   @Test(dependsOnMethods = "testGetPortList")
    public void testCreateFirewallRuleWithPortList() {
-      String id = api().createFirewallRule(networkDomainId, generateFirewallRuleName("server-id"), DEFAULT_ACTION,
+      firewallRuleId = api().createFirewallRule(networkDomainId, generateFirewallRuleName("server-id"), DEFAULT_ACTION,
             DEFAULT_IP_VERSION, DEFAULT_PROTOCOL, FirewallRuleTarget.builder().ip(IpRange.create("ANY", null)).build(),
             FirewallRuleTarget.builder().ip(IpRange.create("ANY", null)).portListId(portListId).build(), Boolean.TRUE,
             Placement.builder().position("LAST").build());
-      firewallRuleIds.add(id);
+      firewallRuleIds.add(firewallRuleId);
+   }
+
+   @Test(dependsOnMethods = "testCreateFirewallRuleWithPortList")
+   public void testListFirewallRules() {
+      PagedIterable<FirewallRule> firewallRulesList = api().listFirewallRules(networkDomainId);
+      assertFalse(firewallRulesList.isEmpty());
+      assertEquals(firewallRulesList.last().get().first().get().networkDomainId(), networkDomainId);
+   }
+
+   @Test(dependsOnMethods = "testListFirewallRules")
+   public void testDeleteFirewallRule() {
+      api().deleteFirewallRule(firewallRuleId);
+      FirewallRule rule = findById(api().listFirewallRules(networkDomainId).concat().toList(), firewallRuleId);
+      assertNull(rule);
    }
 
    @Test(dependsOnMethods = "testDeployNetworkDomain")
    public void testDeployVlan() {
-      vlanId = api().deployVlan(networkDomainId, NetworkApiLiveTest.class.getSimpleName(),
-            NetworkApiLiveTest.class.getSimpleName(), DEFAULT_PRIVATE_IPV4_BASE_ADDRESS,
+      vlanId = api().deployVlan(networkDomainId, NetworkApiLiveTest.class.getSimpleName() + new Date().getTime(),
+            NetworkApiLiveTest.class.getSimpleName() + new Date().getTime(), DEFAULT_PRIVATE_IPV4_BASE_ADDRESS,
             DEFAULT_PRIVATE_IPV4_PREFIX_SIZE);
       assertNotNull(vlanId);
       waitForVlanStatus(api(), vlanId, State.NORMAL, 30 * 60 * 1000, "Error - unable to deploy vlan");
    }
 
    @Test
+   public void testAddPublicIPv4Block() {
+      publicIpv4BlockId = api().addPublicIpBlock(PREPARED_NETWORK_DOMAIN_ID);
+      assertNotNull(publicIpv4BlockId);
+   }
+
+   @Test(dependsOnMethods = "testAddPublicIPv4Block")
+   public void testListPublicIPv4AddressBlocks() {
+      PagedIterable<PublicIpBlock> ipBlockList = api().listPublicIPv4AddressBlocks(PREPARED_NETWORK_DOMAIN_ID);
+      assertTrue(!ipBlockList.isEmpty());
+      assertEquals(ipBlockList.last().get().first().get().size(), 2);
+      assertEquals(ipBlockList.last().get().first().get().networkDomainId(), PREPARED_NETWORK_DOMAIN_ID);
+   }
+
+   @Test(dependsOnMethods = "testListPublicIPv4AddressBlocks")
```

Comment:
This could depend on the "add" test to maximize the chances of this test being executed.